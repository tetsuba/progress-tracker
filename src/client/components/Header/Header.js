import React, { Fragment, useContext } from 'react'
import { Nav, Navbar } from 'react-bootstrap'
import { Link } from 'react-router-dom'

// CONTEXT
// $FlowFixMe - Investigate how to fix this issue
import { AuthenticatedContext } from '../../context/AuthenticatedContext'

export const Header = () => {
  const { authenticated, toggle: userLogout } = useContext(AuthenticatedContext)

  return (
    <Navbar bg="light" expand="lg">
      <Navbar.Brand href="/">Progress Tracker</Navbar.Brand>
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
      <Navbar.Collapse id="basic-navbar-nav">
        <Nav className="mr-auto">
          {!authenticated && (
            <Fragment>
              <Link className="nav-link" to="/">
                Login
              </Link>
              <Link className="nav-link" to="/register">
                Register
              </Link>
            </Fragment>
          )}
          {authenticated && (
            <Fragment>
              <Link className="nav-link" to="/">
                Home
              </Link>
              <Link className="nav-link" to="/students">
                Students
              </Link>
              <Link className="nav-link" to="/myAccount">
                My Account
              </Link>
              <Link
                className="nav-link"
                to="/login"
                onClick={() => userLogout('')}
              >
                Logout
              </Link>
            </Fragment>
          )}
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  )
}
