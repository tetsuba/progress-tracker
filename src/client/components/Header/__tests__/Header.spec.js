import { testRenderer } from '../../../../test/testHelper'
import Header from '../Header'

describe('<Header>', () => {
  it('should render', () => {
    const wrapper = testRenderer(Header, {})
    expect(wrapper).toMatchSnapshot()
  })
})
