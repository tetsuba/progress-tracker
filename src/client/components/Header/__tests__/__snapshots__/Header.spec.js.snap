// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> should render 1`] = `
<Navbar
  bg="light"
  collapseOnSelect={false}
  expand="lg"
  variant="light"
>
  <NavbarBrand
    href="/"
  >
    Progress Tracker
  </NavbarBrand>
  <NavbarToggle
    aria-controls="basic-navbar-nav"
    label="Toggle navigation"
  />
  <NavbarCollapse
    id="basic-navbar-nav"
  >
    <Nav
      className="mr-auto"
      fill={false}
      justify={false}
    >
      <Link
        className="nav-link"
        to="/"
      >
        Login
      </Link>
      <Link
        className="nav-link"
        to="/register"
      >
        Register
      </Link>
    </Nav>
  </NavbarCollapse>
</Navbar>
`;
